skbio/stats/distance/tests/test_base.py

```diff
@@ -113,10 +113,101 @@ def test_init_invalid_input(self):
         with self.assertRaises(DissimilarityMatrixError):
             DissimilarityMatrix(data, [])
 
-        # Non-hollow.
-        data = [[0.0, 1.0], [1.0, 0.01]]
```

Comment:
Can you add the following test cases?

- [ ] test that `DissimilarityMatrix` constructor supports non-hollow data
- [ ] test that `DissimilarityMatrix.from_iterable` supports non-hollow data
- [ ] test that `DissimilarityMatrix.from_iterable` supports asymmetric data
- [ ] test that `DistanceMatrix.from_iterable(validate=True)` is equal to `DistanceMatrix.from_iterable(validate=False)` when passed a valid (i.e., symmetric and hollow) metric
- [ ] test that `DistanceMatrix.from_iterable(validate=False)` "works" with a non-hollow/asymmetric metric (i.e. doesn't raise an error)